algorithms/src/snark/marlin/ahp/indexer/indexer.rs

```diff
@@ -95,24 +95,46 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
             c_arith,
             fft_precomputation,
             ifft_precomputation,
-            mode: PhantomData,
+            id,
+            _mode: PhantomData,
         })
     }
 
-    pub fn index_polynomial_info() -> BTreeMap<PolynomialLabel, PolynomialInfo> {
+    pub fn index_polynomial_info(circuit_ids: Vec<CircuitId>) -> BTreeMap<PolynomialLabel, PolynomialInfo> {
         let mut map = BTreeMap::new();
-        for matrix in ["a", "b", "c"] {
-            map.insert(format!("row_{matrix}"), PolynomialInfo::new(format!("row_{matrix}"), None, None));
-            map.insert(format!("col_{matrix}"), PolynomialInfo::new(format!("col_{matrix}"), None, None));
-            map.insert(format!("val_{matrix}"), PolynomialInfo::new(format!("val_{matrix}"), None, None));
-            map.insert(format!("row_col_{matrix}"), PolynomialInfo::new(format!("row_col_{matrix}"), None, None));
+        for id in circuit_ids {
+            for matrix in ["a", "b", "c"] {
+                map.insert(
+                    format!("circuit_{id}_row_{matrix}"),
```

Comment:
Let's add a variable `let row_label = format!("circuit_{id}_row_{matrix}");` (and similarly for `col`, `row_col`, and `val`), and use those, instead of `repeating  (e.g.)  `format!("circuit_{id}_row_{matrix}")` twice in one `insert` call.

Reply:
In fact we could even use the `index_polynomial_labels` method below to do this cleanly.

Reply:
That is a *lot* better, well spotted! b17840352